packages/one/src/server/oneServe.ts

```diff
@@ -135,6 +135,23 @@ url: ${url}`)
 
   function createHonoHandler(route: RouteInfoCompiled): MiddlewareHandler<BlankEnv, never, {}> {
     return async (context, next) => {
+      if (route.page.endsWith('/+not-found')) {
+        let didCallNext = false
+
+        const response = await serveStatic({
+          root: './dist/client',
+          onFound: (_path, c) => {
+            c.header('Cache-Control', `public, immutable, max-age=31536000`)
```

Comment:
feels like theres a better way to abstract this so it uses the same static header config we have in the other place we use it.